src/gui/widgets/scroll_text.hpp

```diff
@@ -163,6 +165,38 @@ class scroll_text : public scrollbar_container
 
 	void place(const point& origin, const point& size);
 
+	/** See @ref widget::calculate_best_size. */
+	point calculate_best_size() const override
+	{
+		point calc_size = scrollbar_container::calculate_best_size();
+
+		if ((calc_size.x > max_size_.x) && (max_size_.x != 0)) {
+			calc_size.x = max_size_.x;
+		}
+
+		if ((calc_size.y > max_size_.y) && (max_size_.y != 0)) {
+			calc_size.y = max_size_.y;
+		}
+
+		return calc_size;
+	}
+
+	void set_max_size(point max_size)
```

Comment:
While it seems logical to add the padding and scrollbars, please add a comment that this is the max size of the contents, not the container.

Reply:
Added documentation.